### PR TITLE
Add JS bindings Generator test

### LINKED ISSUE #911: Generator JS bindings tests
Implement tests for Generator JS bindings class.

### PR DESCRIPTION
**Pull request by hsorby**
Closes #911.

### CODE REVIEW

**Review by nickerso — DISMISSED** [review]

File: tests/bindings/javascript/generator.test.js [hunk 1]
```diff
@@ -13,3 +13,78 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
+
+        m = p.parseModel(modelString)
+        a = new libcellml.Analyser()
+
+        a.analyseModel(m)
+
+        g.setModel(a.model())
+
+        expect(g.interfaceCode().length).toBe(996)
+        expect(g.implementationCode().length).toBe(1058)
```
Comment (agarny) [inline tests/bindings/javascript/generator.test.js:88]: Shouldn't we just check that we don't have an empty string? I mean, to change the code generator even a tiny bit might break those tests for no good reason.

**Review by agarny — CHANGES_REQUESTED** [review]

File: tests/bindings/javascript/generator.test.js (hunk 1 above)
Comment (nickerso) [reply tests/bindings/javascript/generator.test.js:88]: isn't that the whole point of having tests?

Comment (agarny) [reply tests/bindings/javascript/generator.test.js:88]: For me, the JavaScript tests are here to confirm that what we can do from C++ we can also do from JavaScript, which here means to be able to retrieve some (non-empty) interface/implementation code. The "real" tests are to be done using the C++ implementation.

Comment (hsorby) [reply tests/bindings/javascript/generator.test.js:88]: 792040d

**Review by agarny — DISMISSED** [review]

File: tests/bindings/javascript/generator.test.js (hunk 1 above)
Comment (nickerso) [reply tests/bindings/javascript/generator.test.js:88]: surely that the generated code is accurately being put into JavaScript strings is important to test here? We're already less stringent than the C++ (and Python?) testing by not comparing the string directly.

Is the issue that "standard" developers may not have the build environment to test the JS bindings before making a PR and getting buildbot to run the tests to find if the string has changed length?

Comment (agarny) [reply tests/bindings/javascript/generator.test.js:88]: > Is the issue that "standard" developers may not have the build environment to test the JS bindings before making a PR and getting buildbot to run the tests to find if the string has changed length?

Indeed, libCellML being a C++ library, I would expect "standard" developers to be all set up for C++ development, but not necessarily for Python and/or JavaScript bindings. Now, if we *really* want the test to be "completel" then I would test against the actual expected code, not its length.

Comment (hsorby) [reply tests/bindings/javascript/generator.test.js:88]: Where are we with this?  Do we want the exact line count or not?  I take it from the comments that Andre is keen on the exact count and Alan isn't does that make me the deciding vote?

Comment (agarny) [reply tests/bindings/javascript/generator.test.js:88]: The number of lines doesn't mean anything to me, but I can't be bothered to argue it further, so whatever.

**Review by agarny — APPROVED** [review]

**Review by nickerso — APPROVED** [review]

**Pull request merged**